crates/plex-api/src/media_container/server/library.rs

```diff
@@ -0,0 +1,496 @@
+use std::{fmt, str::FromStr};
+
+use monostate::MustBe;
+use serde::{
+    de::{self, Visitor},
+    Deserialize, Deserializer,
+};
+use serde_aux::prelude::{deserialize_number_from_string, deserialize_option_number_from_string};
+use serde_json::Value;
+use time::{Date, OffsetDateTime};
+
+use crate::media_container::MediaContainer;
+
+fn optional_boolish<'de, D>(deserializer: D) -> Result<Option<bool>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct BoolishVisitor;
+
+    impl<'de> Visitor<'de> for BoolishVisitor {
+        type Value = Option<bool>;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            formatter.write_str("null or something that looks like a boolean")
+        }
+
+        fn visit_none<E>(self) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            Ok(None)
+        }
+
+        fn visit_i64<E>(self, v: i64) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            if v == 0 {
+                Ok(Some(false))
+            } else {
+                Ok(Some(true))
+            }
+        }
+
+        fn visit_u64<E>(self, v: u64) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            if v == 0 {
+                Ok(Some(false))
+            } else {
+                Ok(Some(true))
+            }
+        }
+
+        fn visit_str<E>(self, val: &str) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            if val == "1" {
+                Ok(Some(true))
+            } else {
+                Ok(Some(false))
+            }
+        }
+
+        fn visit_bool<E>(self, v: bool) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            Ok(Some(v))
+        }
+    }
+
+    let visitor = BoolishVisitor;
+    deserializer.deserialize_any(visitor)
+}
+
+#[derive(Debug, Clone, Deserialize, enum_utils::FromStr)]
+#[enumeration(rename_all = "lowercase")]
+#[serde(from = "String")]
+pub enum ChapterSource {
+    Media,
+    Mixed,
+    Agent,
+    #[enumeration(skip)]
+    Unknown(String),
```

Comment:
I love having the string repr of the unknown value inside, and I wanted to do it for the existing enums for a while :D But I'm not sure if always deserializing `Unknown` is a good idea. I'm used to a different approach:
https://github.com/andrey-yantsen/plex-api.rs/blob/22d7a691dcde8882fae412eced82b2319d69b2b2/crates/plex-api/src/media_container/devices.rs#L104-L106

This way, the library will continue working even if Plex adds a new variant (though users won't see the new variant value), but at the same time, we'll notice the new variant when running the tests.

What are your thoughts? I don't have much experience with Rust and easily can be wrong :)

Reply:
Sorry, having difficulty following what you're proposing. Are you saying you'd prefer to just use a plain `Unknown` variant but fail to deserialize it in tests?

Reply:
Let me try rephrasing, as I'm sometimes bad with expressing thoughts in English, as you might've noticed :D

As the end result, I want to have the working library even when Plex adds a new value to an enum. On top of that — I want to have a way for us (the library's developers) to notice if Plex has added a new variant.

If you'll guard your `Unknown(String)` variant with `#[cfg(not(feature = "tests_deny_unknown_fields"))]` it will be enough for me.

My concern here is that I didn't really investigate the best practices on how to achieve the results I want :) So I don't know if what I was doing previously is a good idea at all or if there are any better ways of achieving the same things.

Reply:
Ok. I'm fairly new to Rust too so don't give too much weight to my ideas! I follow you now though and yeah that makes sense. It's also entirely possible I wasn't following because I'm horribly jet-lagged right now!

Reply:
> I'm fairly new to Rust too so don't give too much weight to my ideas!

At least you're working on Firefox, the mother (father?) of all Rust! :) You have better chances to get real-life Rust experience than I right now :)